### DISCUSSION TITLE
Plugins engine

### DISCUSSION THREAD

**Discussion by dkapitanov** [Motion-Project/motion#1513] [General]
I need to extend the standard behavior to handle events. I propose to expand the ability to run scripts by calling the necessary functions from shared libraries. On this basis, it will be possible to make it possible to transfer processing to custom plugins. 
I've prepared a solution and am ready to offer it. 
PS: It may be useful for many tasks like #1177.

**Comment by Mr-DaveDev** [comment]
Based upon the community I am not really considering significant enhancements to the Motion code.

**Comment by dkapitanov** [comment]
I propose only an extension point, it almost does not touch the main code, but it adds a lot of possibilities. For details https://github.com/dkapitanov/motion/tree/plugins

**Reply by DiagnozHub** [reply]
It has ability  to preporocess every frame from video?
For exaple get frame as image do thomething (detect objects maybe) and continue usual work.
